### DISCUSSION TITLE
`prefer-no-csd` as a window rule property

### DISCUSSION THREAD

**Discussion by zeroeightysix** [YaLTeR/niri#800] [Ideas]
I'm not against CSD, so don't really want to disable them as the toplevel, but would for example prefer `alacritty` not to have a title bar. For this reason I tried seeing if `prefer-no-csd` was available as a window-rule:

```
window-rule {
    match app-id="^Alacritty$"

    prefer-no-csd
}
```

but it isn't. If this seems like a sensible change, I'd be happy trying to PR it!

**Comment by YaLTeR** [comment]
The reason it works as it does is because SDL2 for a long while had a bug causing windows to never open if the compositor had the xdg-decoration global but used it to ask the client for CSD. There are now lots of games bundling SDL2 versions with this bug. So when prefer-no-csd is unset, niri completely hides xdg-decoration from clients. 

What could maybe work is having a separate client-rule section that matches on things associated with a process id (e.g. executable name) and can selectively enable the xdg-decoration global to trigger SSD for select clients or at runtime.

**Reply by zeroeightysix** [reply]
for the SDL bug: how does sway handle this? Maybe I just don't play enough games, but I don't recall ever having to think about window decorations or running into a bug like that on sway.

for the `client-rule` suggestion: That does sound interesting, and perhaps opens up new possibilities for other applications as well! Can this also work in the opposite way, offering `xdg-decoration` to all clients _but not_ a select few as defined by a `client-rule`?

**Reply by YaLTeR** [reply]
> how does sway handle this?

Don't know, but sway asks for SSD by default, which is not affected.

> Can this also work in the opposite way, offering xdg-decoration to all clients but not a select few as defined by a client-rule?

With window rules you can use exclude to get an inverted rule, same principle will apply here.

**Comment by alexdev16** [comment]
Alacritty can turn off its own decorations. Open alacritty.toml and add the line 'decorations = "None"' to the section [window].

**Reply by zeroeightysix** [reply]
Thanks, didn't know :)
Will be using this!
